inductiva/_benchmarks/benchmark.py

```diff
@@ -0,0 +1,119 @@
+"""API for Benchmarking"""
+import json
+from typing import Self
+from inductiva import types
+from inductiva.simulators.simulator import Simulator
+from inductiva.projects.project import Project
+from inductiva.resources import machine_groups
+
+
+class Benchmark(Project):
+    """Represents the benchmark runner."""
+
+    def __init__(self, name: str, append: bool = True):
+        """
+        Initializes a new Benchmark instance.
+
+        Args:
+            name (str): The name of the benchmark runner.
+            append (bool): Indicates whether to allow adding runs to the 
+            existing benchmark (default is True).
+        """
+        super().__init__(name=name, append=append)
+        self.runs = []
+
+    def add_run(
+        self,
+        simulator: Simulator,
+        input_dir: str,
+        on: types.ComputationalResources,
+        **kwargs,
+    ) -> Self:
+        """
+        Adds a simulation run to the benchmark.
+
+        Args:
+            simulator (Simulator): The simulator to be used for this run.
+            input_dir (str): The directory containing input files for the
+            simulation.
+            on (types.ComputationalResources): The computational resources to
+            run the simulation on.
+            **kwargs: Additional keyword arguments for the simulator run.
+
+        Returns:
+            Self: The current instance for method chaining.
+        """
+        self.runs.append((simulator, input_dir, on, kwargs))
+        return self
+
+    def run(self, num_repeats: int = 2) -> Self:
+        """
+        Executes all added runs.
+
+        Each run is executed the specified number of times, and the collection
+        of runs is cleared afterwards.
+
+        Args:
+            num_repeats (int): The number of times to repeat each simulation
+            run (default is 2).
+
+        Returns:
+            Self: The current instance for method chaining.
+        """
+        with self:
+            for simulator, input_dir, machine_group, kwargs in self.runs:
+                machine_group.start(wait_on_pending_quota=True)
+                for _ in range(num_repeats):
+                    simulator.run(input_dir=input_dir,
+                                  on=machine_group,
+                                  **kwargs)
+            self.runs.clear()
+        return self
+
+    def wait(self) -> Self:
+        """
+        Waits for all running tasks to complete.
+
+        Returns:
+            Self: The current instance for method chaining.
+        """
+        tasks = self.get_tasks()
+        for task in tasks:
+            task.wait(download_std_on_completion=False)
+        return self
+
+    def to_dict(self):
```

Comment:
What would happen if I called to_dict before a benchmark finishes running?
Maybe we should only collect info about tasks that are successful?  What about tasks that are still running?

We need to make a decision here. To_dict can only be called if all tasks are finished (in terminal state).
What if some tasks fail?

Reply:
Currently, it should fail -- tasks that are not in a terminal state don't have ```info```, right? Even so, the ```wait``` method was implemented to prevent this issue from happening.

I think we should collect information about tasks that are in a terminal state, even if there's a failure.

By the way, this method has been replaced by ```gather_metrics``` and ```gather_detailed_data``` in the most recent PR.